### ISSUE TITLE
HRRR stations list issue [BUG]

### ISSUE THREAD

**Issue by sssangha** [dbekaert/RAiDER#563]
**Describe the bug**
The `raider.py` program fails when I pass a list of station tropo zenith delays as an argument for station_field encompassing CONUS, as it spans beyond the available range of `HRRR` data. Looks like we will have to adjust the padding behavior [here](https://github.com/dbekaert/RAiDER/blob/dev/tools/RAiDER/models/weatherModel.py#L206) but am not sure how best to proceed. Perhaps previous ticket #552 is of relevance?

**To Reproduce**
Obtain the GNSS tropo zenith delays:
raiderDownloadGNSS.py --out CONUS_00 --date 20090101 20191231 12 --returntime '00:00:00' --bounding_box '21 49 -133 -60' --cpus 8

Using the attached GNSS tropo zenith delays `gnssStationList_overbbox.csv` + `yaml` file, run raider:
raider.py raider_stations.yaml

**Desktop (please complete the following information):**
 - RAiDER git tag: 9ddd4abad8414b3670194b4c08220f150d7ba00d
 - OS

[gnssStationList_overbbox.csv](https://github.com/dbekaert/RAiDER/files/12009196/gnssStationList_overbbox.csv)
[raider_stations.yaml.txt](https://github.com/dbekaert/RAiDER/files/12009200/raider_stations.yaml.txt)

**Comment by sssangha** [comment]
Warning log:
```
Extent of the weather model is (xmin, ymin, xmax, ymax):-134.08, 21.14, -60.97, 47.83
Extent of the input is (xmin, ymin, xmax, ymax): -125.60, 20.96, -59.96, 49.06
ERROR: The weather model passed does not cover all of the input points; you may need to download a larger area.
ERROR: No weather model data was successfully obtained.
Traceback (most recent call last):
  File "/u/leffe-data/ssangha/conda_installation/stable_july30_2020/envs/RAiDER/bin/raider.py", line 8, in <module>
    sys.exit(main())
  File "/u/leffe-data/ssangha/conda_installation/stable_july30_2020/RAiDER/tools/RAiDER/cli/__main__.py", line 42, in main
    process_entry_point.load()()
  File "/u/leffe-data/ssangha/conda_installation/stable_july30_2020/RAiDER/tools/RAiDER/cli/raider.py", line 276, in calcDelays
    raise RuntimeError
RuntimeError
```

**Comment by bbuzz31** [comment]
While this  message is slightly wrong (fixed in #564), this error is valid. The bounding box of the stations requested is indeed outside of the weather model. In the figure, the blue is the extent of the HRRR, the black dots are the GNSS stations requested, and the red outline is the bounding box that encompasses the stations. 

![image](https://github.com/dbekaert/RAiDER/assets/15522503/5746f68d-1495-4449-9a5b-b0eb76e31a76)

**Comment by bbuzz31** [comment]
@sssangha proposed internally truncating the station file list if they extend beyond the HRRR box. 
This would require some dedicated code that I'm not sure is worth implementing; this would seem to be an error on the users side.

What if we instead adjust `get_station_list` in `downloadGNSSdelays.py` to accept a shapefile? 
Thoughts @sssangha @jlmaurer @dbekaert?

**Comment by jlmaurer** [comment]
Hey @bbuzz31 @sssangha I think we should handle this type of issue by ignoring it. i.e., if points are passed that fall outside the bounding box the code should just proceed as normal (while raising the warning) and the delays for those points will end up as NaNs. I don't think this is a problem as it occurs strictly in the last step (3D interpolation). This is the most intuitive way to handle this as you don't lose any stations (which could be a problem for users) and you don't calculate delays for stations that are outside the box.

**Comment by bbuzz31** [comment]
Closed with #564
